Rocket.Chat/Views/Avatar/AvatarView.swift

```diff
@@ -24,7 +24,13 @@ final class AvatarView: UIView {
 
     var user: User? {
         didSet {
-            updateAvatar()
+            if let user = user {
+                if !user.isInvalidated {
+                    updateAvatar()
+                }
+            } else {
+                updateAvatar()
+            }
```

Comment:
How about we revert this and change the userAvatarURL() function instead?
That way we don't crash if we call it from somewhere else.

```swift
private func userAvatarURL() -> URL? {
    guard let user = user, !user.isInvalidated,
          let username = user.username,
          let auth = AuthManager.isAuthenticated(),
          let baseURL = auth.baseURL(),
          let encodedUsername = username.addingPercentEncoding(withAllowedCharacters: CharacterSet.urlQueryAllowed)
    else {
        return nil
    }

    return URL(string: "\(baseURL)/avatar/\(encodedUsername)")
}
```

Reply:
@cardoso The problem with that is that we also use `user` object when we need to set the initials.

Reply:
@rafaelks I'd just check there too. I think it's more maintainable if we add the check as pre-condition to the functions instead of before calling them, that way we can never forget.

Also would be better if we made the functions in this class more testable like
```swift 
func avatarWithInitialsForString(_ string: String) -> (backgroundColor: UIColor, text: String)? {
    ... // don't need user here
}
```
But that's just a suggestion, we can refactor this class later :)